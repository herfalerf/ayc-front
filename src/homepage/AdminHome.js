import React from "react";
import { Grid, Typography } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";

// Test component for admin homepage
//
// routed at /admin/home
//
// Routes -> AdminHome

const useStyles = makeStyles((theme) => ({
  root: { flexGrow: 1 },
  upper: {
    backgroundImage: `url('${process.env.PUBLIC_URL}/images/admin-home/admin-home01.png')`,
    backgroundSize: "cover",
    backgroundPosition: "center bottom",
    minHeight: "25vh",
  },
  adminHeader: {
    padding: theme.spacing(2),
  },
}));

const AdminHome = () => {
  console.debug("AdminHome");

  const classes = useStyles();

  return (
    <div>
      <div className={classes.upper}></div>
      <Grid container justifyContent="center" className={classes.adminHeader}>
        <Grid item xs={12}>
          <Typography variant="h4" align="center">
            {" "}
            Welcome to the Admin Portal
          </Typography>
        </Grid>
        <Grid item xs={8}>
          <Typography variant="subtitle1" align="center">
            You may now use the admin drop down menu on the main navigation bar
            to access admin specific pages or to log out. Please be aware that
            changes you make on the admin portal will permanently affect the
            rest of the site.
          </Typography>
        </Grid>
      </Grid>
    </div>
  );
};

export default AdminHome;
