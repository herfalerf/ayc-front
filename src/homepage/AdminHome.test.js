import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import AdminHome from "./AdminHome";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <AdminHome />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <AdminHome />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
