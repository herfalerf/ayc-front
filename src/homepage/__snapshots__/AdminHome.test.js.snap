// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-upper-5"
    />
    <div
      class="MuiGrid-root makeStyles-adminHeader-6 MuiGrid-container MuiGrid-justify-content-xs-center"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
        >
           Welcome to the Admin Portal
        </h4>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter"
        >
          You may now use the admin drop down menu on the main navigation bar to access admin specific pages or to log out. Please be aware that changes you make on the admin portal will permanently affect the rest of the site.
        </h6>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
